File: client/src/app/models/product.ts
export interface Product {
  id: number;
  name: string;
  ingredients: string;
  price: number;
  pictureUrl: string;
  grams: number;
  calories: number;
  proteins: number;
  carbohidrates: number;
  fats: number;
  quantityInStock?: number;
  sideDish?: string;
  meat: string;
}

export interface ProductParams {
  orderBy: string;
  searchTerm?: string;
  meats: string[];
  sideDishes: string[];
  pageNumber: number;
  pageSize: number;
}

File: API/Entities/Product.cs
namespace API.Entities;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Ingredients { get; set; }
    public long Price { get; set; }
    public string PictureUrl { get; set; }
    public string SideDish { get; set; }
    public string Meat { get; set; }
    public long Grams { get; set; }
    public long Calories { get; set; }
    public long Proteins { get; set; }
    public long Carbohidrates { get; set; }
    public long Fats { get; set; }
    public int QuantityInStock { get; set; }
    public string PublicId { get; set; }
}
